### PR TITLE
Fixes for advanced tabs form logic

### PR DESCRIPTION
**Pull request by zomars**
## What does this PR do?

- Adds disabled cursor for disabled switches
- Fixes logic for seated events

https://user-images.githubusercontent.com/3504472/192757971-749b3f1c-6d58-47a9-8040-641732168aa7.mov

**Environment**: Staging(main branch) / Production

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Chore (refactoring code, technical debt, workflow improvements)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How should this be tested?

- [ ] Go to an event type edit view and play with the advanced type values. Should be working as expected

### CODE REVIEW

File: apps/web/components/v2/eventtype/EventAdvancedTab.tsx [hunk 1]
```diff
@@ -42,7 +42,6 @@ export const EventAdvancedTab = ({ eventType, team }: Pick<EventTypeSetupInfered
   const [hashedLinkVisible, setHashedLinkVisible] = useState(!!eventType.hashedLink);
   const [hashedUrl, setHashedUrl] = useState(eventType.hashedLink?.link);
   const [seatsInputVisible, setSeatsInputVisible] = useState(!!eventType.seatsPerTimeSlot);
-  const [seatsPerTimeSlot, setSeatsPerTimeSlot] = useState(eventType.seatsPerTimeSlot || 2);
```
Comment (zomars) [inline apps/web/components/v2/eventtype/EventAdvancedTab.tsx:45]: When using form, we should not rely on `useState` as much as possible. We already have a "Form State" we should take advantage of it as much as possible.

File: apps/web/components/v2/eventtype/EventAdvancedTab.tsx [hunk 2]
```diff
@@ -313,45 +312,57 @@ export const EventAdvancedTab = ({ eventType, team }: Pick<EventTypeSetupInfered
       />
       <hr />
       <Controller
-        name="seatsPerTimeSlot"
+        name="seatsPerTimeSlotEnabled"
```
Comment (zomars) [inline apps/web/components/v2/eventtype/EventAdvancedTab.tsx:315]: Added a pseudo-field that won't be send to the backend but will allow us to keep track of this state locally.

File: apps/web/components/v2/eventtype/EventAdvancedTab.tsx [hunk 3]
```diff
@@ -313,45 +312,57 @@ export const EventAdvancedTab = ({ eventType, team }: Pick<EventTypeSetupInfered
...
+              onCheckedChange={(e) => {
+                setSeatsInputVisible(e);
+                // Enabling seats will disable guests and requiring confimation until fully supported
+                if (e) {
+                  formMethods.setValue("disableGuests", true);
+                  formMethods.setValue("requiresConfirmation", false);
+                  formMethods.setValue("seatsPerTimeSlot", 2);
+                } else {
+                  formMethods.setValue("seatsPerTimeSlot", null);
+                }
```
Comment (zomars) [inline apps/web/components/v2/eventtype/EventAdvancedTab.tsx:332]: Here's the actual logic, we don't need to manipulate external state, just the actual form values.

File: apps/web/components/v2/eventtype/EventAdvancedTab.tsx [hunk 4]
```diff
@@ -313,45 +312,57 @@ export const EventAdvancedTab = ({ eventType, team }: Pick<EventTypeSetupInfered
...
+            <div className="flex flex-col">
+              <Label className="text-sm font-semibold leading-none text-black">{t("offer_seats")}</Label>
+              <p className="-mt-2 text-sm leading-normal text-gray-600">{t("offer_seats_description")}</p>
+            </div>
+          </div>
+        )}
+      />
+      {seatsInputVisible && (
+        <Controller
+          name="seatsPerTimeSlot"
```
Comment (zomars) [inline apps/web/components/v2/eventtype/EventAdvancedTab.tsx:346]: The Switch (which is a boolean) was manipulating a number value. This makes it right.

File: packages/ui/v2/core/Switch.tsx [hunk 5]
```diff
@@ -21,7 +21,8 @@ const Switch = (
     <div className={classNames("flex h-auto w-auto flex-row items-center", props.fitToHeight && "h-fit")}>
       <PrimitiveSwitch.Root
         className={classNames(
-          props.checked ? "bg-gray-900" : "bg-gray-200 hover:bg-gray-300",
+          props.checked ? "bg-gray-900" : "bg-gray-200",
+          primitiveProps.disabled ? "cursor-not-allowed" : "hover:bg-gray-300",
```
Comment (zomars) [inline packages/ui/v2/core/Switch.tsx:25]: At least for desktop users, disabled switches will display a "non allowed" cursor. Tried adding opacity as well but it becomes confusion between on-disabled/off-disabled/on-enabled/off-enabled cc @Jaibles any suggestions for mobile users?

**Review by emrysal — APPROVED** [review]
I don't really like pseudo-fields, at that point you're better off with state again, which Controller under the hood also does. Minor nit.

**Comment by zomars** [comment]
> I don't really like pseudo-fields, at that point you're better off with state again, which Controller under the hood also does. Minor nit.

We could make it an actual field and move this logic to the backend tho. "If input. seatsPerTimeSlotEnabled is false then set the seatsPerTimeSlot to null"

**Comment by emrysal** [comment]
> We could make it an actual field and move this logic to the backend tho. "If input. seatsPerTimeSlotEnabled is false then set the seatsPerTimeSlot to null"

I'd say no as it's purely FE logic, if I ask myself "What if I update through the API" that doesn't make sense. But asking myself the same question the following logic does come to mind as belonging to the BE.

```javascript
// Enabling seats will disable guests and requiring confirmation until fully supported
if (e) {
  formMethods.setValue("disableGuests", true);
  formMethods.setValue("requiresConfirmation", false);
  formMethods.setValue("seatsPerTimeSlot", 2);
} else {
  formMethods.setValue("seatsPerTimeSlot", null);
}
```

**Pull request merged**